### PR TITLE
add kudo addon

### PR DESCRIPTION
**Pull request by s12chung**
tested with `konvoy up`. the controller deploys fine.

ran:

```
❯ kubectl kudo install kafka --instance=kafka
instance.kudo.dev/v1alpha1/kafka created
```

created the resources, however, `kafka` isn't running properly. I think that's out of scope.

for reference, this is what I got:

```
❯ kubectl describe pod kafka-kafka-0
Name:           kafka-kafka-0
Namespace:      default
Priority:       0
Node:           ip-10-0-131-128.us-west-2.compute.internal/10.0.131.128
Start Time:     Tue, 22 Oct 2019 13:03:45 -0400
Labels:         app=kafka
                controller-revision-hash=kafka-kafka-bd9dfb9dd
                heritage=kudo
                kafka=kafka
                kudo.dev/instance=kafka
                kudo.dev/operator=kafka
                statefulset.kubernetes.io/pod-name=kafka-kafka-0
Annotations:    cni.projectcalico.org/podIP: 192.168.122.215/32
                kudo.dev/operator-version: 0.2.0
                kudo.dev/phase: deploy-kafka
                kudo.dev/plan: deploy
                kudo.dev/plan-execution: 
                kudo.dev/step: deploy
Status:         Running
IP:             192.168.122.215
IPs:            <none>
Controlled By:  StatefulSet/kafka-kafka
Containers:
  k8skafka:
    Container ID:  containerd://91adfa632adfefa378ec0c51d5b1b0a0bf8e427a11e6275bcf22a8c63dd28f11
    Image:         mesosphere/kafka:0.2.0-2.3.0
    Image ID:      docker.io/mesosphere/kafka@sha256:e941829a4da5672e83ce5bdff33f4d92469625ae8b5a73cc1d6e34d73459c700
    Port:          9093/TCP
    Host Port:     0/TCP
    Command:
      sh
      -c
      exec $KAFKA_HOME/bin/start-kafka.sh
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Tue, 22 Oct 2019 13:05:23 -0400
      Finished:     Tue, 22 Oct 2019 13:05:32 -0400
    Ready:          False
    Restart Count:  3
    Limits:
      cpu:     2
      memory:  4Gi
    Requests:
      cpu:      500m
      memory:   2Gi
    Liveness:   exec [sh -c $KAFKA_HOME/readiness-check.sh] delay=30s timeout=1s period=30s #success=1 #failure=3
    Readiness:  exec [sh -c $KAFKA_HOME/readiness-check.sh] delay=0s timeout=30s period=10s #success=1 #failure=3
    Environment:
      KAFKA_HEAP_OPTS:       -Xmx512M -Xms512M
      KAFKA_OPTS:            -Dlogging.level=INFO
      METRICS_OPTS:          -javaagent:/opt/jmx-exporter/jmx_prometheus_javaagent.jar=9094:/metrics/metrics.properties
      KAFKA_ZK_URI:          zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181/kafka
      GC_LOG_ENABLED:        false
      LOG_DIR:               /var/lib/kafka/data
      METRICS_ENABLED:       true
      KAFKA_CLIENT_ENABLED:  true
      KAFKA_BROKER_PORT:     9093
      KAFKA_CLIENT_PORT:     9092
    Mounts:
      /config from config (rw)
      /metrics from metrics (rw)
      /var/lib/kafka from datadir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-qzm8w (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  datadir:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  datadir-kafka-kafka-0
    ReadOnly:   false
  config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      kafka-serverproperties
    Optional:  false
  metrics:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      kafka-metrics-config
    Optional:  false
  default-token-qzm8w:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-qzm8w
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason                  Age                  From                                                 Message
  ----     ------                  ----                 ----                                                 -------
  Normal   Scheduled               2m27s                default-scheduler                                    Successfully assigned default/kafka-kafka-0 to ip-10-0-131-128.us-west-2.compute.internal
  Normal   SuccessfulAttachVolume  2m25s                attachdetach-controller                              AttachVolume.Attach succeeded for volume "pvc-1243caa0-4e9d-4f0d-b059-1164cb1a62de"
  Warning  Unhealthy               102s (x2 over 112s)  kubelet, ip-10-0-131-128.us-west-2.compute.internal  Readiness probe errored: rpc error: code = Unknown desc = failed to exec in container: container is in CONTAINER_EXITED state
  Normal   Pulling                 50s (x4 over 2m9s)   kubelet, ip-10-0-131-128.us-west-2.compute.internal  Pulling image "mesosphere/kafka:0.2.0-2.3.0"
  Normal   Pulled                  50s (x4 over 2m1s)   kubelet, ip-10-0-131-128.us-west-2.compute.internal  Successfully pulled image "mesosphere/kafka:0.2.0-2.3.0"
  Normal   Created                 49s (x4 over 2m1s)   kubelet, ip-10-0-131-128.us-west-2.compute.internal  Created container k8skafka
  Normal   Started                 49s (x4 over 2m1s)   kubelet, ip-10-0-131-128.us-west-2.compute.internal  Started container k8skafka
  Warning  Unhealthy               42s (x2 over 72s)    kubelet, ip-10-0-131-128.us-west-2.compute.internal  Readiness probe failed:
  Warning  BackOff                 40s (x5 over 102s)   kubelet, ip-10-0-131-128.us-west-2.compute.internal  Back-off restarting failed container
```

```
❯ kubectl logs kafka-kafka-0 
starting the kafka broker using broker.id 0...
[2019-10-22 17:06:15,499] INFO Registered kafka:type=kafka.Log4jController MBean (kafka.utils.Log4jControllerRegistration$)
[2019-10-22 17:06:15,995] INFO Registered signal handlers for TERM, INT, HUP (org.apache.kafka.common.utils.LoggingSignalHandler)
[2019-10-22 17:06:15,996] INFO starting (kafka.server.KafkaServer)
[2019-10-22 17:06:15,997] INFO Connecting to zookeeper on zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181/kafka (kafka.server.KafkaServer)
[2019-10-22 17:06:16,024] INFO [ZooKeeperClient Kafka server] Initializing a new session to zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181. (kafka.zookeeper.ZooKeeperClient)
[2019-10-22 17:06:16,028] INFO Client environment:zookeeper.version=3.4.14-4c25d480e66aadd371de8bd2fd8da255ac140bcf, built on 03/06/2019 16:18 GMT (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,028] INFO Client environment:host.name=kafka-kafka-0.kafka-svc.default.svc.cluster.local (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,028] INFO Client environment:java.version=1.8.0_212 (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,028] INFO Client environment:java.vendor=AdoptOpenJDK (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,028] INFO Client environment:java.home=/usr/share/java (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,029] INFO Client environment:java.class.path=/opt/kafka/bin/../libs/activation-1.1.1.jar:/opt/kafka/bin/../libs/aopalliance-repackaged-2.5.0.jar:/opt/kafka/bin/../libs/argparse4j-0.7.0.jar:/opt/kafka/bin/../libs/audience-annotations-0.5.0.jar:/opt/kafka/bin/../libs/commons-lang3-3.8.1.jar:/opt/kafka/bin/../libs/connect-api-2.3.0.jar:/opt/kafka/bin/../libs/connect-basic-auth-extension-2.3.0.jar:/opt/kafka/bin/../libs/connect-file-2.3.0.jar:/opt/kafka/bin/../libs/connect-json-2.3.0.jar:/opt/kafka/bin/../libs/connect-runtime-2.3.0.jar:/opt/kafka/bin/../libs/connect-transforms-2.3.0.jar:/opt/kafka/bin/../libs/guava-20.0.jar:/opt/kafka/bin/../libs/hk2-api-2.5.0.jar:/opt/kafka/bin/../libs/hk2-locator-2.5.0.jar:/opt/kafka/bin/../libs/hk2-utils-2.5.0.jar:/opt/kafka/bin/../libs/jackson-annotations-2.9.9.jar:/opt/kafka/bin/../libs/jackson-core-2.9.9.jar:/opt/kafka/bin/../libs/jackson-databind-2.9.9.jar:/opt/kafka/bin/../libs/jackson-dataformat-csv-2.9.9.jar:/opt/kafka/bin/../libs/jackson-datatype-jdk8-2.9.9.jar:/opt/kafka/bin/../libs/jackson-jaxrs-base-2.9.9.jar:/opt/kafka/bin/../libs/jackson-jaxrs-json-provider-2.9.9.jar:/opt/kafka/bin/../libs/jackson-module-jaxb-annotations-2.9.9.jar:/opt/kafka/bin/../libs/jackson-module-paranamer-2.9.9.jar:/opt/kafka/bin/../libs/jackson-module-scala_2.12-2.9.9.jar:/opt/kafka/bin/../libs/jakarta.annotation-api-1.3.4.jar:/opt/kafka/bin/../libs/jakarta.inject-2.5.0.jar:/opt/kafka/bin/../libs/jakarta.ws.rs-api-2.1.5.jar:/opt/kafka/bin/../libs/javassist-3.22.0-CR2.jar:/opt/kafka/bin/../libs/javax.servlet-api-3.1.0.jar:/opt/kafka/bin/../libs/javax.ws.rs-api-2.1.1.jar:/opt/kafka/bin/../libs/jaxb-api-2.3.0.jar:/opt/kafka/bin/../libs/jersey-client-2.28.jar:/opt/kafka/bin/../libs/jersey-common-2.28.jar:/opt/kafka/bin/../libs/jersey-container-servlet-2.28.jar:/opt/kafka/bin/../libs/jersey-container-servlet-core-2.28.jar:/opt/kafka/bin/../libs/jersey-hk2-2.28.jar:/opt/kafka/bin/../libs/jersey-media-jaxb-2.28.jar:/opt/kafka/bin/../libs/jersey-server-2.28.jar:/opt/kafka/bin/../libs/jetty-client-9.4.18.v20190429.jar:/opt/kafka/bin/../libs/jetty-continuation-9.4.18.v20190429.jar:/opt/kafka/bin/../libs/jetty-http-9.4.18.v20190429.jar:/opt/kafka/bin/../libs/jetty-io-9.4.18.v20190429.jar:/opt/kafka/bin/../libs/jetty-security-9.4.18.v20190429.jar:/opt/kafka/bin/../libs/jetty-server-9.4.18.v20190429.jar:/opt/kafka/bin/../libs/jetty-servlet-9.4.18.v20190429.jar:/opt/kafka/bin/../libs/jetty-servlets-9.4.18.v20190429.jar:/opt/kafka/bin/../libs/jetty-util-9.4.18.v20190429.jar:/opt/kafka/bin/../libs/jopt-simple-5.0.4.jar:/opt/kafka/bin/../libs/jsr305-3.0.2.jar:/opt/kafka/bin/../libs/kafka-clients-2.3.0.jar:/opt/kafka/bin/../libs/kafka-log4j-appender-2.3.0.jar:/opt/kafka/bin/../libs/kafka-streams-2.3.0.jar:/opt/kafka/bin/../libs/kafka-streams-examples-2.3.0.jar:/opt/kafka/bin/../libs/kafka-streams-scala_2.12-2.3.0.jar:/opt/kafka/bin/../libs/kafka-streams-test-utils-2.3.0.jar:/opt/kafka/bin/../libs/kafka-tools-2.3.0.jar:/opt/kafka/bin/../libs/kafka_2.12-2.3.0-sources.jar:/opt/kafka/bin/../libs/kafka_2.12-2.3.0.jar:/opt/kafka/bin/../libs/log4j-1.2.17.jar:/opt/kafka/bin/../libs/lz4-java-1.6.0.jar:/opt/kafka/bin/../libs/maven-artifact-3.6.1.jar:/opt/kafka/bin/../libs/metrics-core-2.2.0.jar:/opt/kafka/bin/../libs/osgi-resource-locator-1.0.1.jar:/opt/kafka/bin/../libs/paranamer-2.8.jar:/opt/kafka/bin/../libs/plexus-utils-3.2.0.jar:/opt/kafka/bin/../libs/reflections-0.9.11.jar:/opt/kafka/bin/../libs/rocksdbjni-5.18.3.jar:/opt/kafka/bin/../libs/scala-library-2.12.8.jar:/opt/kafka/bin/../libs/scala-logging_2.12-3.9.0.jar:/opt/kafka/bin/../libs/scala-reflect-2.12.8.jar:/opt/kafka/bin/../libs/slf4j-api-1.7.26.jar:/opt/kafka/bin/../libs/slf4j-log4j12-1.7.26.jar:/opt/kafka/bin/../libs/snappy-java-1.1.7.3.jar:/opt/kafka/bin/../libs/spotbugs-annotations-3.1.9.jar:/opt/kafka/bin/../libs/validation-api-2.0.1.Final.jar:/opt/kafka/bin/../libs/zkclient-0.11.jar:/opt/kafka/bin/../libs/zookeeper-3.4.14.jar:/opt/kafka/bin/../libs/zstd-jni-1.4.0-1.jar:/opt/jmx-exporter/jmx_prometheus_javaagent.jar (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,029] INFO Client environment:java.library.path=/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,029] INFO Client environment:java.io.tmpdir=/tmp (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,029] INFO Client environment:java.compiler=<NA> (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,029] INFO Client environment:os.name=Linux (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,029] INFO Client environment:os.arch=amd64 (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,029] INFO Client environment:os.version=3.10.0-957.1.3.el7.x86_64 (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,029] INFO Client environment:user.name=kafka (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,029] INFO Client environment:user.home=/home/kafka (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,029] INFO Client environment:user.dir=/opt/kafka (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,030] INFO Initiating client connection, connectString=zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181 sessionTimeout=6000 watcher=kafka.zookeeper.ZooKeeperClient$ZooKeeperClientWatcher$@1e1a0406 (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:16,040] INFO [ZooKeeperClient Kafka server] Waiting until connected. (kafka.zookeeper.ZooKeeperClient)
[2019-10-22 17:06:16,046] WARN Session 0x0 for server zk-zookeeper-0.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-0.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:16,153] WARN Session 0x0 for server zk-zookeeper-2.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-2.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:16,257] WARN Session 0x0 for server zk-zookeeper-1.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-1.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:17,358] WARN Session 0x0 for server zk-zookeeper-0.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-0.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:17,464] WARN Session 0x0 for server zk-zookeeper-2.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-2.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:17,567] WARN Session 0x0 for server zk-zookeeper-1.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-1.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:18,667] WARN Session 0x0 for server zk-zookeeper-0.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-0.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:18,768] WARN Session 0x0 for server zk-zookeeper-2.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-2.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:18,869] WARN Session 0x0 for server zk-zookeeper-1.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-1.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:19,969] WARN Session 0x0 for server zk-zookeeper-0.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-0.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:20,070] WARN Session 0x0 for server zk-zookeeper-2.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-2.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:20,171] WARN Session 0x0 for server zk-zookeeper-1.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-1.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:21,272] WARN Session 0x0 for server zk-zookeeper-0.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-0.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:21,372] WARN Session 0x0 for server zk-zookeeper-2.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-2.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:21,473] WARN Session 0x0 for server zk-zookeeper-1.zk-hs:2181, unexpected error, closing socket connection and attempting reconnect (org.apache.zookeeper.ClientCnxn)
java.lang.IllegalArgumentException: Unable to canonicalize address zk-zookeeper-1.zk-hs:2181 because it's not resolvable
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:65)
	at org.apache.zookeeper.SaslServerPrincipal.getServerPrincipal(SaslServerPrincipal.java:41)
	at org.apache.zookeeper.ClientCnxn$SendThread.startConnect(ClientCnxn.java:1001)
	at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1060)
[2019-10-22 17:06:22,042] INFO [ZooKeeperClient Kafka server] Closing. (kafka.zookeeper.ZooKeeperClient)
[2019-10-22 17:06:22,676] INFO Session: 0x0 closed (org.apache.zookeeper.ZooKeeper)
[2019-10-22 17:06:22,676] INFO EventThread shut down for session: 0x0 (org.apache.zookeeper.ClientCnxn)
[2019-10-22 17:06:22,679] INFO [ZooKeeperClient Kafka server] Closed. (kafka.zookeeper.ZooKeeperClient)
[2019-10-22 17:06:22,682] ERROR Fatal error during KafkaServer startup. Prepare to shutdown (kafka.server.KafkaServer)
kafka.zookeeper.ZooKeeperClientTimeoutException: Timed out waiting for connection while in state: CONNECTING
	at kafka.zookeeper.ZooKeeperClient.$anonfun$waitUntilConnected$3(ZooKeeperClient.scala:258)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
	at kafka.utils.CoreUtils$.inLock(CoreUtils.scala:253)
	at kafka.zookeeper.ZooKeeperClient.waitUntilConnected(ZooKeeperClient.scala:254)
	at kafka.zookeeper.ZooKeeperClient.<init>(ZooKeeperClient.scala:112)
	at kafka.zk.KafkaZkClient$.apply(KafkaZkClient.scala:1826)
	at kafka.server.KafkaServer.createZkClient$1(KafkaServer.scala:364)
	at kafka.server.KafkaServer.$anonfun$initZkClient$2(KafkaServer.scala:381)
	at kafka.server.KafkaServer.$anonfun$initZkClient$2$adapted(KafkaServer.scala:379)
	at scala.Option.foreach(Option.scala:274)
	at kafka.server.KafkaServer.initZkClient(KafkaServer.scala:379)
	at kafka.server.KafkaServer.startup(KafkaServer.scala:207)
	at kafka.server.KafkaServerStartable.startup(KafkaServerStartable.scala:38)
	at kafka.Kafka$.main(Kafka.scala:84)
	at kafka.Kafka.main(Kafka.scala)
[2019-10-22 17:06:22,683] INFO shutting down (kafka.server.KafkaServer)
[2019-10-22 17:06:22,689] INFO shut down completed (kafka.server.KafkaServer)
[2019-10-22 17:06:22,690] ERROR Exiting Kafka. (kafka.server.KafkaServerStartable)
[2019-10-22 17:06:22,697] INFO shutting down (kafka.server.KafkaServer)
```

### CODE REVIEW

File: templates/kudo.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+apiVersion: kubeaddons.mesosphere.io/v1alpha1
+kind: Addon
+metadata:
+  name: kudo
+  namespace: kubeaddons
+  labels:
+    kubeaddons.mesosphere.io/name: kudo
+  annotations:
+    catalog.kubeaddons.mesosphere.io/addon-revision: "0.0.1-1"
```
Comment (shaneutt) [inline templates/kudo.yaml]:
```suggestion
    catalog.kubeaddons.mesosphere.io/addon-revision: "0.7.4-1"
```

File: templates/kudo.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    appversion.kubeaddons.mesosphere.io/kudo: "0.7.4"
+spec:
+  namespace: kube-system
+  kubernetes:
+    minSupportedVersion: v1.15.0
```
Comment (shaneutt) [inline templates/kudo.yaml]:
```suggestion
    minSupportedVersion: v1.16.0
```

File: templates/kudo.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      enabled: true
+    - name: docker
+      enabled: true
+    - name: none
+      enabled: true
+  chartReference:
+    chart: kudo
+    repo: https://mesosphere.github.io/charts/staging
+    version: 0.1.1
+    values: |
```
Comment (shaneutt) [inline templates/kudo.yaml]:
```suggestion
```

File: templates/kudo.yaml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      ---
```
Comment (shaneutt) [inline templates/kudo.yaml]:
```suggestion
```

**Comment by alejandroEsc** [comment]
at first glance looks like a zookeeper issue? Either its required and not started or not set up correctly? I dont think it should block this PR though, would give it a lgtm if when we get this to succeed.

**Comment by shaneutt** [comment]
> at first glance looks like a zookeeper issue? Either its required and not started or not set up correctly? I dont think it should block this PR though, would give it a lgtm if when we get this to succeed.

Roger, if @s12chung marks it "ready" and we've tested this for our purposes I wont block, just waiting on some other changes above that I am blocking on.

File: templates/kudo.yaml (hunk 1 above)
Comment (s12chung) [reply templates/kudo.yaml]: done.

File: templates/kudo.yaml (hunk 2 above)
Comment (s12chung) [reply templates/kudo.yaml]: done.

File: templates/kudo.yaml (hunk 3 above)
Comment (s12chung) [reply templates/kudo.yaml]: done.

File: templates/kudo.yaml (hunk 4 above)
Comment (s12chung) [reply templates/kudo.yaml]: done.

**Comment by s12chung** [comment]
@shaneutt 

> created the resources, however, kafka isn't running properly. ****I think that's out of scope.****

emphasis on above :p

**Pull request merged**